### PR TITLE
User new login

### PR DESCRIPTION
**Pull request by scmountain**
Rspec 4-0-1

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,3 +1,8 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+  helper_method :current_user
+
+  def current_user
+    @user ||= User.find(session[:user_id]) if session[:user_id]
+  end
```
Comment (CjMoore) [inline app/controllers/application_controller.rb]: 🔥

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  def new
+  end
+
+  def create
+
+    @user = User.find_by(username: params[:session][:username])
+    if @user && @user.authenticate(params[:session][:password])
+      session[:user_id] = @user.id
+      admin_authenticate
```
Comment (CjMoore) [inline app/controllers/sessions_controller.rb]: Whats this line doing?

Comment (scmountain) [inline app/controllers/sessions_controller.rb]: whoops, :)

File: db/migrate/20170329202812_create_users.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class CreateUsers < ActiveRecord::Migration[5.0]
+  def change
+    create_table :users do |t|
```
Comment (CjMoore) [inline db/migrate/20170329202812_create_users.rb]: Nice migration yo.

File: spec/factories/users.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+FactoryGirl.define do
+  factory :user do
+    first_name "MyString"
```
Comment (CjMoore) [inline spec/factories/users.rb]: Do we want to use faker for this?

File: spec/features/00_user_new_login_spec.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+               phone: "555-555-5555",
+               email: "billygoat@gmail.com",
+              )
+   visit login_path
+
+  fill_in "session[username]", with: "billygoat"
+  fill_in "session[password]", with: "pass"
+
+   expect(current_path).to eq(login_path)
+ end
```
Comment (CjMoore) [inline spec/features/00_user_new_login_spec.rb]: Solid test.

File: spec/factories/users.rb (hunk 4 above)
Comment (scmountain) [inline spec/factories/users.rb]: hand rolled my user

Comment (CjMoore) [inline spec/factories/users.rb]: Cool. I think its a good idea to go with whatever you're most comfortable with for now. We can standardize our test setups with factory girl when we refactor.

Comment (cews7) [inline spec/factories/users.rb]: that sounds like a good plan.

**Pull request merged**